va/x11/va_x11.c

```diff
@@ -77,8 +77,37 @@ static VAStatus va_DisplayContextGetDriverNames(
 {
     VAStatus vaStatus = VA_STATUS_ERROR_UNKNOWN;
 
-    if (!getenv("LIBVA_DRI3_DISABLE"))
+    const char* dri3_disable = getenv("LIBVA_DRI3_DISABLE");
+    if (!dri3_disable || !atoi(dri3_disable)) {
+        unsigned old_num_drivers = *num_drivers;
+
         vaStatus = va_DRI3_GetDriverNames(pDisplayContext, drivers, num_drivers);
+        /* As of 8 July 2023, i965 and iHD drivers lack DRI3 support.
+         *
+         * Requests by the community were raised as early as 29 July 2017,
+         * with DRI3 support landing in libva on the 27 September 2022. At of
+         * time of writing it's unknown if/when that would materialise.
+         *
+         * To handle this on libva level, we are explicitly disabling DRI3
+         * support on said drivers - it scales better than having every user
+         * to set the environment override listed above.
+         *
+         * Omit them by default, set LIBVA_DRI3_DISABLE=0 to bypass.
+         */
+        if (vaStatus == VA_STATUS_SUCCESS && dri3_disable && !atoi(dri3_disable)) {
```

Comment:
The above condition skips the Intel check when LIBVA_DRI_DISABLE=0 is set, so the behavior is reverse from intended (intended: LIBVA_DRI_DISABLE=0 allows intel dri3, unset variable denies intel dri3).

The following would work as intended (tested), i.e. automatic intel skipping only done if the variable is unset:

    if (vaStatus == VA_STATUS_SUCCESS && !dri3_disable) {